to.etc.domui/src/main/java/to/etc/domui/state/IWithStoredState.java

```diff
@@ -0,0 +1,10 @@
+package to.etc.domui.state;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+@NonNullByDefault
+public interface IWithStoredState {
```

Comment:
We do not use Object as parameter type anymore. Use <T> on the interface and let T replace Object.